### PR TITLE
DSNS 117-Introduction of an equivalent process for repackaging landsat data as the current one gets decommission from crontab/flock.

### PR DESCRIPTION
**Pull request by gordonyeong**
DSNS 117-Introduction of an equivalent process for repackaging
landsat data as the current one gets decommission from crontab/flock.

### CODE REVIEW

File: dags/nci_ard/nci_landsat_repackage.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+# kick off running fetch on NCI
+with dag:
+    start = DummyOperator(task_id="start")
+    completed = DummyOperator(task_id="completed")
+
+    COMMON = """
+        #  ts_nodash timestamp no dashes.
+        {% set log_ext = ts_nodash %}
```
Comment (dunkgray) [inline dags/nci_ard/nci_landsat_repackage.py:93]:
```suggestion
```

File: dags/nci_ard/nci_landsat_repackage.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        """
+
+    submit_task_id = "submit_downloader"
+    submit_downloader = SSHOperator(
+        task_id=submit_task_id,
+        command=COMMON
+        + """
+mkdir -p {{ params.logs_dir }}{{ log_ext }};
```
Comment (dunkgray) [inline dags/nci_ard/nci_landsat_repackage.py:101]:
```suggestion
mkdir -p {{ params.logs_dir }}
```

File: dags/nci_ard/nci_landsat_repackage.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+mkdir -p {{ params.logs_dir }}archive/;
+{{ params.downloader_script }} {{ params.mode }} {{ params.logs_dir }}{{ log_ext }} \
+>> {{ params.logs_dir }}{{ log_ext }}/{{ params.log_file }} \
```
Comment (dunkgray) [inline dags/nci_ard/nci_landsat_repackage.py:104]:
```suggestion
```

File: dags/nci_ard/nci_landsat_repackage.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+2>> {{ params.logs_dir }}{{ log_ext }}/{{ params.error_file }} ; \
```
Comment (dunkgray) [inline dags/nci_ard/nci_landsat_repackage.py:105]:
```suggestion
```

File: dags/nci_ard/nci_landsat_repackage.py (hunk 3 above)
Comment (gordonyeong) [reply dags/nci_ard/nci_landsat_repackage.py:104]: this is following the other airflowjobs where they redirect the output to a log file and error from stderr to an error file. 
Getting rid of line 104 would mean the former will not happen.

File: dags/nci_ard/nci_landsat_repackage.py (hunk 4 above)
Comment (gordonyeong) [reply dags/nci_ard/nci_landsat_repackage.py:105]: this is following the other airflowjobs where they redirect the output to a log file and error from stderr to an error file.
Getting rid of line 104 would mean the latter will not happen.

File: dags/nci_ard/nci_landsat_repackage.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        """
+
+    submit_task_id = "submit_downloader"
+    submit_downloader = SSHOperator(
+        task_id=submit_task_id,
+        command=COMMON
+        + """
+mkdir -p {{ params.logs_dir }}{{ log_ext }};
+mkdir -p {{ params.logs_dir }}archive/;
+{{ params.downloader_script }} {{ params.mode }} {{ params.logs_dir }}{{ log_ext }} \
```
[the remark this answers is not in the document]
Comment (gordonyeong) [reply dags/nci_ard/nci_landsat_repackage.py:103]: if we dont have a log_ext, i suppose we will not be able to find out when the run was done aside from querying the contents of the logfile.
 It also means that log contents of previous runs will be lost. 
 This is not consistent with the other dags in the directory.
 
 Is this what we want to do?

**Review by dunkgray — APPROVED** [review]
Looks good
One suggestion - move the log file usgs-c2.jsonl out of the timestamped dir since the shell script is dealing with rotating this log

**Review by dunkgray — APPROVED** [review]
looks good

**Pull request merged**